Tests/PlayerTests/FeatureFlagProviderTests.swift

```diff
@@ -4,40 +4,70 @@ import EventProducer
 import XCTest
 
 final class FeatureFlagProviderTests: XCTestCase {
-	override func setUpWithError() throws {
-		PlayerWorld = PlayerWorldClient.mock
+	private var featureFlagProvider: FeatureFlagProvider!
+	private var shouldUseOfflineEngine: Bool = false
+
+	override func setUp() {
 		Player.shared = nil
+
+		featureFlagProvider = FeatureFlagProvider.mock
+		featureFlagProvider.isOfflineEngineEnabled = {
+			self.shouldUseOfflineEngine
+		}
 	}
 
 	func testOffliningIsNotInitialized() throws {
-		PlayerWorld.developmentFeatureFlagProvider.isOffliningEnabled = false
+		shouldUseOfflineEngine = false
 
 		let playerInstance = Player.bootstrap(
 			listener: PlayerListenerMock(),
 			listenerQueue: DispatchQueue(label: "com.tidal.queue.for.testing"),
-			featureFlagProvider: .mock,
+			featureFlagProvider: featureFlagProvider,
 			credentialsProvider: CredentialsProviderMock(),
 			eventSender: EventSenderMock()
 		)
 
 		let player = try XCTUnwrap(playerInstance)
 		XCTAssertNotNil(player.playerEngine)
+		XCTAssertNil(player.offlineStorage)
 		XCTAssertNil(player.offlineEngine)
 	}
 
 	func testOffliningIsInitialized() throws {
-		PlayerWorld.developmentFeatureFlagProvider.isOffliningEnabled = true
+		shouldUseOfflineEngine = true
 
 		let playerInstance = Player.bootstrap(
 			listener: PlayerListenerMock(),
 			listenerQueue: DispatchQueue(label: "com.tidal.queue.for.testing"),
-			featureFlagProvider: .mock,
+			featureFlagProvider: featureFlagProvider,
 			credentialsProvider: CredentialsProviderMock(),
 			eventSender: EventSenderMock()
 		)
 
 		let player = try XCTUnwrap(playerInstance)
 		XCTAssertNotNil(player.playerEngine)
+		XCTAssertNotNil(player.offlineStorage)
+		XCTAssertNotNil(player.offlineEngine)
+	}
+
+	func testOffliningIsInitializedAfterBootstrap() throws {
```

Comment:
[suggestion] is we start with false, then would it make sense to assert nil before we change? And same if going from false to true

Reply:
In my mind, that case is already covered in the test `func testOffliningIsNotInitialized() throws`

I guess both could be combined.